pkg/core/object/replicate.go

```diff
@@ -59,6 +62,9 @@ func EncodeReplicationMetaInfo(cID cid.ID, oID, firstPart, previousPart oid.ID,
 	if len(locked) > 0 {
 		kvs = append(kvs, oidsKV(lockedKey, locked))
 	}
+	if typ != objectsdk.TypeRegular {
+		kvs = append(kvs, kv(typeKey, uint32(typ)))
```

Comment:
Notice that tombstone/lock can be detected by deleted/locked already. But this covers more things, so OK.

Reply:
you mean lock type with some non-empy `deleted` should be an error (panic in fact) in this func?